vscode-lean4/src/utils/batch.ts

```diff
@@ -16,16 +16,18 @@ export async function batchExecute(
             options = { cwd: workingDirectory };
         }
 
-        // The mocha test framework listens to process.on('uncaughtException')
-        // which is raised if spawn cannot find the command and the test automatically
-        // fails with "Uncaught Error: spawn elan ENOENT".  Therefore we manually
-        // check if the command exists so as not to trigger that exception.
-
         try {
-            const fullPath = findProgramInPath(executablePath);
-            if (!fullPath) {
-                resolve(undefined);
-                return;
+            if (isRunningTest())
+            {
+                // The mocha test framework listens to process.on('uncaughtException')
```

Comment:
I understand now. So the question is, could we turn this behaviour off in Mocha and delete the whole `findProgramInPath` check? If not, let's go with the `isRunningTest` check.

Reply:
Hm, but why is the exception raised by `spawn` uncaught? Maybe that's the problem in the first place?

Reply:
I don't think so, the mocha code that registers `uncaughtException` is buried pretty deep in mocha, and I couldn't find a way to disable it or undo it.

Reply:
> Hm, but why is the exception raised by `spawn` uncaught? Maybe that's the problem in the first place?

yeah, it's weird I know, we have a try/catch which catches it, but spawn still raises it as a process.uncaughtException for some reason, and this is then causing mocha to fail the test.  It is all very weird.

Reply:
PS: this code was already there by the way, I'm just making it test only code now.